// config/editProfile.js
var models = require('config/models');


exports.editEmail = function(phone_number,email,callback) {

	models.User.find({phone_number: phone_number},function(err,users){

		if(users.length != 0){

			users[0].email = email;
			users[0].save();
		}
	});

	callback({'response' : "Email changed successfully"});

}

exports.editUserName = function(phone_number,user_name,callback) {

	models.User.find({phone_number: phone_number},function(err,users){

		if(users.length != 0){

			users[0].user_name = user_name;
			users[0].save();
		}
	});

	callback({'response' : "User Name changed successfully"});

}

exports.editImage = function(phone_number,image,callback) {

	models.User.find({phone_number: phone_number},function(err,users){

		if(users.length != 0){

			users[0].image = image;
			users[0].save();
		}
	});

	callback({'response' : "Image changed successfully"});

}

exports.editPhoneNumber = function(phone_number,new_phone_number,callback) {

	models.User.find({phone_number: phone_number},function(err,users){
		


		if(users.length != 0){

			console.log(phone_number);
			console.log(new_phone_number);
			users[0].phone_number = new_phone_number;
			users[0].save();
		}
	});

	callback({'response' : "phone number changed successfully"});

}

exports.editSharedLocation = function(phone_number,shared_location,callback) {



	models.User.find({phone_number: phone_number},function(err,users){
		console.log(phone_number);
		console.log(shared_location);

		if(users.length != 0){

			//console.log(users[0].shared_location);
			//console.log(shared_location);
			users[0].shared_location = shared_location;
			users[0].save();
		}
	});

	callback({'response' : "Shared Location updated"});

}

exports.editAllowedToPost = function(phone_number,allowed_to_post,callback) {

	models.User.find({phone_number: phone_number},function(err,users){

		if(users.length != 0){

			users[0].allowed_to_post = allowed_to_post;
			users[0].save();
		}
	});

	callback({'response' : "Allowed_to_post changed successfully"});

}
